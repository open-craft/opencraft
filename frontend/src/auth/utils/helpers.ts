import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
