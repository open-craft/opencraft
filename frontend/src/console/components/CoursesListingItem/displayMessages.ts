const messages = {
  starts: {
    defaultMessage: 'Starts: ',
    description: ''
  }
};

export default messages;
