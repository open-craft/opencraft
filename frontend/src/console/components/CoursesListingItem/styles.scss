.courses-listing-item {
  border: 1px solid #bebebe;
  border-bottom: 1.5px solid #126f9a;
  padding-left: 50px;
  flex-direction: column;
  height: 140px;
  max-width: 100px;

  .col {
    height: auto;
    padding: 0 0 0 5px;
    font-size: 10px;
    flex-grow: unset;
    color: #474747;
  }

  .course-image {
    padding-left: 0;
    padding-bottom: 6px;

    img {
      max-width: 100%;
    }
  }

  .organization, .course-number {
    font-size: 6px;
  }

  .course-name {
    .customizable-link {
      height: auto;
      display: contents;
      padding: 0;
      color: #0075b4;
      span {
        font-size: 7px;
      }
    }
  }

  .course-start {
    padding-bottom: 3px;
    flex-grow: 1;
    display: inline-flex;
    align-items: flex-end;
    font-size: 6px;
  }
}
