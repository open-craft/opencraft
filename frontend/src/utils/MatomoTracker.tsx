import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
