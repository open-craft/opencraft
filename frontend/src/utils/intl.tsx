import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
