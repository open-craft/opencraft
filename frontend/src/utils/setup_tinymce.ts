import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
