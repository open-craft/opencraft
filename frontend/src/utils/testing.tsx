import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
