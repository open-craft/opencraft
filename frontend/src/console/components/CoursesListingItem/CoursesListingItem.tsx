import * as React from 'react';
import './styles.scss';
import { Col, Row } from 'react-bootstrap';
import { ThemeSchema } from 'ocim-client';
import courseImage from 'assets/course_image.jpg';
import messages from './displayMessages';
import { WrappedMessage } from '../../../utils/intl';
import { CustomizableLink } from '../CustomizableLink';

interface CoursesListingItemProps {
  themeData: ThemeSchema;
  loggedIn?: boolean;
}

export const CoursesListingItem: React.FC<CoursesListingItemProps> = (
  props: CoursesListingItemProps
) => {
  const { themeData } = props;
  const textStyle = { color: themeData.mainColor };
  // TODO: Make text customizable.

  return (
    <Row
      className="courses-listing-item"
      style={{ borderBottomColor: themeData.linkColor }}
    >
      <Col className="course-image">
        <img src={courseImage} alt="course" />
      </Col>
      <Col className="organization" style={textStyle}>
        edX
      </Col>
      <Col className="course-number" style={textStyle}>
        DemoX
      </Col>
      <Col className="course-name">
        <CustomizableLink linkColor={themeData.linkColor} noHover>
          <span>Demonstration Course</span>
        </CustomizableLink>
      </Col>
      <Col className="course-start" style={textStyle}>
        <WrappedMessage messages={messages} id="starts" />
        Feb 5, 2013
      </Col>
    </Row>
  );
};
