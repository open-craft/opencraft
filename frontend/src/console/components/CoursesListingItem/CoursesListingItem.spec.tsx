import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { CoursesListingItem } from './CoursesListingItem';

it('renders without crashing', () => {
    const tree = setupComponentForTesting(<CoursesListingItem  themeData={{}}/>).toJSON();
    expect(tree).toMatchSnapshot();
});
