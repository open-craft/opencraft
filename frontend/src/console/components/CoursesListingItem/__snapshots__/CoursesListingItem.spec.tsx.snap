// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="courses-listing-item row"
  style={
    Object {
      "borderBottomColor": undefined,
    }
  }
>
  <div
    className="course-image col"
  >
    <img
      alt="course"
      src="course_image.jpg"
    />
  </div>
  <div
    className="organization col"
    style={
      Object {
        "color": undefined,
      }
    }
  >
    edX
  </div>
  <div
    className="course-number col"
    style={
      Object {
        "color": undefined,
      }
    }
  >
    DemoX
  </div>
  <div
    className="course-name col"
  >
    <button
      className="customizable-link noHover"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "borderBottomColor": undefined,
          "color": undefined,
        }
      }
      type="button"
    >
      <span>
        Demonstration Course
      </span>
    </button>
  </div>
  <div
    className="course-start col"
    style={
      Object {
        "color": undefined,
      }
    }
  >
    Starts: 
    Feb 5, 2013
  </div>
</div>
`;
